ocp_resources/resource.py

```diff
@@ -385,6 +387,15 @@ def __init__(
         self.teardown = teardown
         self.timeout = timeout
         self.delete_timeout = delete_timeout
+        self.node_selector = node_selector
+        self.node_selector_labels = node_selector_labels
+        self._node_selector = self._set_node_selector()
```

Comment:
why do we need both `self.node_selector` and `self._node_selector` ? this is confusing

Reply:
self.node_selector is the one that we got from the use.
self._node_selector is what we set in the body {f"{self.ApiGroup.KUBERNETES_IO}/hostname": self.node_selector}

Agree, needs to be renamed